src/App.tsx

```diff
@@ -1,18 +1,39 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const total = 42;
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const onPageChange = (page: number) => (
+    setCurrentPage(page)
+  );
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const lastItem = currentPage * perPage > total
+    ? total
+    : currentPage * perPage;
+  const firstItem = (currentPage * perPage) - perPage + 1;
+  const itemsToShow = items.slice(firstItem - 1, lastItem);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstItem} - ${lastItem} of ${total})`}
```

Comment:
you don't need to Template literals here, because React already has data binding for that, like:
Page { currentPage } ( items { firstItem } - { lastItem } of { total }  )

Reply:
Can't make it this way, cause linter say to plays every part of sentence on a new line, and it's less readable(((